spec/polaris-management-service.yml

```diff
@@ -850,9 +850,92 @@ components:
         - $ref: "#/components/schemas/Catalog"
         - type: object
           properties:
-            remoteUrl:
-              type: string
-              description: URL to the remote catalog API
+            connectionConfigInfo:
+              $ref: "#/components/schemas/ConnectionConfigInfo"
+
+    ConnectionConfigInfo:
+      type: object
+      description: A connection configuration representing a remote catalog service
+      properties:
+        connectionType:
+          type: string
+          enum:
+            - ICEBERG_REST
+          description: The type of remote catalog service represented by this connection
+        remoteUri:
+          type: string
+          description: URI to the remote catalog service
+      required:
+        - connectionType
+      discriminator:
+        propertyName: connectionType
+        mapping:
+          ICEBERG_REST: "#/components/schemas/IcebergRestConnectionConfigInfo"
+
+    IcebergRestConnectionConfigInfo:
+      type: object
+      description: Configuration necessary for connecting to an Iceberg REST Catalog
+      allOf:
+        - $ref: '#/components/schemas/ConnectionConfigInfo'
+      properties:
+        remoteCatalogName:
+          type: string
+          description: The name of a remote catalog instance within the remote catalog service; in some older systems
+            this is specified as the 'warehouse' when multiple logical catalogs are served under the same base
+            remoteUri, and often translates into a 'prefix' added to all REST resource paths
+        restAuthentication:
+          $ref: "#/components/schemas/RestAuthenticationInfo"
+
+    RestAuthenticationInfo:
+      type: object
+      description: Authentication-specific information for a REST connection
+      properties:
+        restAuthenticationType:
+          type: string
+          enum:
+            - OAUTH
+            - BEARER
+          description: The type of authentication to use when connecting to the remote rest service
+      required:
+        - restAuthenticationType
+      discriminator:
+        propertyName: restAuthenticationType
+        mapping:
+          OAUTH: "#/components/schemas/OauthRestAuthenticationInfo"
+          BEARER: "#/components/schemas/BearerRestAuthenticationInfo"
+
+    OauthRestAuthenticationInfo:
+      type: object
+      description: OAuth authentication based on client_id/client_secret
```

Comment:
If it's just about the client credentials flow, maybe we can rename the type to be something like `OAuthClientCredentialsParameters`?.. Iceberg REST Servers do not have to be restricted to client credentials, other flows may be available (e.g. delegation), which are still OAuth2, but will require different parameters.

Reply:
Done, including the casing `OAuth` to match conventions elsewhere.